restock.lic

```diff
@@ -67,7 +67,7 @@ class Restock
     if coin_needed > 0
       ### The following line is for the bribe to the night attendant to let you in.
       coin_needed += 2502 if UserVars.sun['night'] && town =~ /\bShard\b/
-      DRCM.get_money_from_bank("#{coin_needed} copper", "#{DRCM.town_currency(town)}", "#{town}")
+      DRCM.ensure_copper_on_hand(coin_needed, DRCM.town_currency(town), "#{town}")
```

Comment:
I don't think this needs the concatenation, can use the `town` variable directly?
```suggestion
      DRCM.ensure_copper_on_hand(coin_needed, DRCM.town_currency(town), town)
```

Reply:
I can try that. It is working as expected the way it is. I made a runestone a lot of money to test withdrawal of large amounts of money. I will remove #{} and see.

Reply:
> I can try that. It is working as expected the way it is. I made a runestone a lot of money to test withdrawal of large amounts of money. I will remove #{} and see.
`DRCM.ensure_copper_on_hand(coin_needed, DRCM.town_currency(town), town)`:
```
2021-07-11 13:06:30 -0400:--- Lich: error: undefined method `bankbot_enabled' for "dokoras":String
2021-07-11 13:06:30 -0400:	common-money:85:in `withdraw_exact_amount?'
2021-07-11 13:06:30 -0400:	common-money:79:in `block in ensure_copper_on_hand'
```
`DRCM.ensure_copper_on_hand(coin_needed, DRCM.town_currency(town), "#{town}")`:
```
2021-07-11 13:11:22 -0400:Sierack nods and says, "I ask 8118 copper dokoras or if you would prefer, 8 gold, 1 silver, 1 bronze and 8 copper."
2021-07-11 13:11:22 -0400:[restock]>offer 8118
2021-07-11 13:11:22 -0400:You feel fully rested.
etc...
```

Reply:
Thanks for checking. That is beyond strange. From the usage I thought for sure `town` is already a string variable. Seems odd the code behaves differently when the string variable is used as-is vs redundantly converted into a string again. Whelp, ignore my suggestion then!

Reply:
Is `town` nil by chance? The string interpolation would convert the nil to an empty string, which doesn't trigger the check on line 73 in `common-money`. This sounds like a bad argument being passed to `ensure_copper_on_hand` that happens to work.

Reply:
Sierack is in Shard, so Shard was the town passed from the read of the runestones:
```
  Refresh:
    hometown: Shard
    name: quartz runestone
    room: 13105
    price: 8118
    size: 1
    stackable: false
    quantity: 6
    container: *zill_pouch
```

It is coming as a string, I believe, from line 32 in restock and passing to the town variable in `def restock_items`.

Reply:
Can you do a puts/message/echo/etc and check the value both with and without the string interpolation?

Reply:
I updated the line. The results are below. 

```
>;restock
--- Lich: restock active.
[sanowret-crystal]>exhale my sanowret crystal
You exhale softly on your sanowret crystal, and scintillating sparks of light dance across its surface.
You hear in your mind a quiet recollection of wisdom.  Though you think that you could get more knowledge if you were to GAZE into the crystal, you come away from the experience with a further understanding of Arcana as the scintillating lights fade again.
[restock]>open my embroidered backpack
That is already open.
[restock]>open my party sack
That is already open.
[restock]>open my watery portal
[Assuming you mean a swirling eddy of incandescent light bound by a gold-striated coralite frame.]
You can't do that.
[restock]>open my green backpack
That is already open.
[restock]>open my blue backpack
That is already open.
[restock]>open my red backpack
That is already open.
[restock]>open my duffel bag
That is already open.
[restock]>open my zillinen pouch
That is already open.
[restock]>open my thigh bag
That is already open.
[restock]>open my champion's belt
That is already open.
[restock]>open my wyvern-hide lootpouch
That is already open.
| Crossing
[restock]>count my first boar-tusk arrows
You count some boar-tusk arrows in the quiver and see there are ninety left.
[restock]>count my second boar-tusk arrows
You count some boar-tusk arrows in the quiver and see there are sixty-eight left.
[restock]>count my third boar-tusk arrows
I could not find what you were referring to.
[restock]>count my first crossbow bolts
You count some crossbow bolts in the quiver and see there are one hundred fifty left.
[restock]>count my second crossbow bolts
You count some crossbow bolts in the quiver and see there are three left.
[restock]>count my third crossbow bolts
I could not find what you were referring to.
| Hibarnhvidar
[restock]>count my first smooth rocks
You count some smooth rocks in the quiver and see there are ninety-three left.
[restock]>count my second smooth rocks
I could not find what you were referring to.
| Shard
[restock]>rummage /C runestone IN MY zillinen pouch
[restock]>rummage /C runestone IN MY zillinen pouch
[restock]>rummage /C runestone IN MY zillinen pouch
You rummage through an extraordinary zillinen pouch with perfect proportions looking for something similar to "runestone":
runestones (44): 
    topaz runestone (4)
    pale avaes runestone (5)
    quartz runestone (11)
    colorful elbaite runestone (5)
    electrum runestone (6)
    fiery rhodonite runestone
    glossy axinite runestone (5)
    dark azurite runestone (5)
    ilmenite runestone (2)
[restock]>rummage /C runestone IN MY zillinen pouch
[restock]>rummage /C runestone IN MY zillinen pouch
You rummage through an extraordinary zillinen pouch with perfect proportions looking for something similar to "runestone":
runestones (44): 
    topaz runestone (4)
    pale avaes runestone (5)
    quartz runestone (11)
    colorful elbaite runestone (5)
    electrum runestone (6)
    fiery rhodonite runestone
    glossy axinite runestone (5)
    dark azurite runestone (5)
    ilmenite runestone (2)
[restock]>rummage /C runestone IN MY zillinen pouch
[restock]>rummage /C runestone IN MY zillinen pouch
You rummage through an extraordinary zillinen pouch with perfect proportions looking for something similar to "runestone":
runestones (44): 
    topaz runestone (4)
    pale avaes runestone (5)
    quartz runestone (11)
    colorful elbaite runestone (5)
    electrum runestone (6)
    fiery rhodonite runestone
    glossy axinite runestone (5)
    dark azurite runestone (5)
    ilmenite runestone (2)
| Shard
[restock]>wealth dokoras
You rummage through an extraordinary zillinen pouch with perfect proportions looking for something similar to "runestone":
runestones (44): 
    topaz runestone (4)
    pale avaes runestone (5)
    quartz runestone (11)
    colorful elbaite runestone (5)
    electrum runestone (6)
    fiery rhodonite runestone
    glossy axinite runestone (5)
    dark azurite runestone (5)
    ilmenite runestone (2)
  No Dokoras.
| Shard
You rummage through an extraordinary zillinen pouch with perfect proportions looking for something similar to "runestone":
runestones (44): 
    topaz runestone (4)
    pale avaes runestone (5)
    quartz runestone (11)
    colorful elbaite runestone (5)
    electrum runestone (6)
    fiery rhodonite runestone
    glossy axinite runestone (5)
    dark azurite runestone (5)
    ilmenite runestone (2)
You rummage through an extraordinary zillinen pouch with perfect proportions looking for something similar to "runestone":
runestones (44): 
    topaz runestone (4)
    pale avaes runestone (5)
    quartz runestone (11)
    colorful elbaite runestone (5)
    electrum runestone (6)
    fiery rhodonite runestone
    glossy axinite runestone (5)
    dark azurite runestone (5)
    ilmenite runestone (2)
--- Lich: go2 active.

--- Lich: go2 has exited.
[restock]>withdraw 2 platinum
The clerk counts out 2 platinum Dokoras and hands them over, making a notation in her ledger.
| Shard
[restock]>withdraw 3 gold
The clerk counts out 3 gold Dokoras and hands them over, making a notation in her ledger.
| Shard
[restock]>withdraw 4 silver
The clerk counts out 4 silver Dokoras and hands them over, making a notation in her ledger.
| Shard
[restock]>withdraw 5 bronze
The clerk counts out 5 bronze Dokoras and hands them over, making a notation in her ledger.
| Shard
[restock]>withdraw 2 copper
The clerk counts out 2 copper Dokoras and hands them over, making a notation in her ledger.
--- Lich: go2 active.

--- Lich: go2 has exited.
[restock]>buy avaes runestone
Sierack nods and says, "I ask 4510 copper dokoras or if you would prefer, 4 gold, 5 silver and 1 bronze."
[restock]>offer 4510
Sierack lifts a pale avaes runestone from a shelf and holds it up.  After giving it one final inspection he hands it to you.
[restock]>put my avaes runestone in my zillinen pouch
You put your runestone in your zillinen pouch.
[restock]>buy topaz runestone
Sierack nods and says, "I ask 8569 copper dokoras or if you would prefer, 8 gold, 5 silver, 6 bronze and 9 copper."
[restock]>offer 8569
Sierack lifts a topaz runestone from a shelf and holds it up.  After giving it one final inspection he hands it to you.
[restock]>put my topaz runestone in my zillinen pouch
You put your runestone in your zillinen pouch.
[restock]>buy topaz runestone
Sierack nods and says, "I ask 8569 copper dokoras or if you would prefer, 8 gold, 5 silver, 6 bronze and 9 copper."
[restock]>offer 8569
Sierack lifts a topaz runestone from a shelf and holds it up.  After giving it one final inspection he hands it to you.
[restock]>put my topaz runestone in my zillinen pouch
You put your runestone in your zillinen pouch.
[restock]>buy elbaite runestone
Sierack nods and says, "I ask 1804 copper dokoras or if you would prefer, 1 gold, 8 silver and 4 copper."
[restock]>offer 1804
Sierack lifts a colorful elbaite runestone from a shelf and holds it up.  After giving it one final inspection he hands it to you.
[restock]>put my elbaite runestone in my zillinen pouch
You put your runestone in your zillinen pouch.
--- Lich: go2 active.

--- Lich: go2 has exited.
Swigglewhip and the clerk share a few words.
[restock]>wealth
Wealth:
  3 silver Kronars (300 copper Kronars).
  No Lirums.
  No Dokoras.
[restock]>deposit all
The shifting winds blowing from behind you die down.
You don't have any Dokoras to deposit.
```